lm_eval/utils.py

```diff
@@ -218,23 +249,38 @@ def complete_code(
         with torch.no_grad():
             if task.stop_words:
                 # Set the start_length after which to check for stopping to be the longest input ignoring padding
-                gen_kwargs["stopping_criteria"][0].start_length = (
-                    batch["input_len"].max().item()
-                )
+                max_len = batch["input_len"].max().item()
+                if "ids_encoder" in batch:
+                    max_len += 1 # Add 1 for decoder_start_token_id
+                gen_kwargs["stopping_criteria"][0].start_length = max_len
+            if hasattr(task, "max_length_multiplier") and task.max_length_multiplier:
+                idx = 1 if task.stop_words else 0
+                gen_kwargs["stopping_criteria"][idx].input_length = batch["input_len"].max().item()                
+            
             inputs = batch["ids"][:, : batch["input_len"]]
-            if is_wrapped:
-                # 8bit and 4bit models are wrapped in accelerator
-                generated_tokens = accelerator.unwrap_model(model).generate(
-                    input_ids=inputs,
-                    num_return_sequences=batch_size,
-                    **gen_kwargs,
-                )
-            else:
+            if "ids_encoder" in batch:
                 generated_tokens = model.generate(
```

Comment:
this assumes the seq2seq models won't be wrapped (loaded in 8bit..)

Reply:
Nice fixed!